init docs for `HTMLElement.virtualkeyboardpolicy` (#31367)
mdn/content: files/en-us/web/api/htmlelement/virtualkeyboardpolicy/index.md

---
title: "HTMLElement: virtualKeyboardPolicy property"
short-title: virtualKeyboardPolicy
slug: Web/API/HTMLElement/virtualKeyboardPolicy
page-type: web-api-instance-property
status:
  - experimental
browser-compat: api.HTMLElement.virtualKeyboardPolicy
---

{{APIRef("VirtualKeyboard API")}}{{SeeCompatTable}}

> The **`virtualKeyboardPolicy`** property of the {{domxref("HTMLElement")}} interface gets and sets a string indicating the on-screen virtual keyboard behavior on devices such as tablets, mobile phones, or other devices where a hardware keyboard may not be available, if the element is contenteditable (like {{htmlelement("input")}}, {{htmlelement("textarea")}} or the [`contenteditable`](/en-US/docs/Web/HTML/Global_attributes/contenteditable) set on the element).

Reviewer: ```suggestion
The **`virtualKeyboardPolicy`** property of the {{domxref("HTMLElement")}} interface gets and sets a string indicating the on-screen virtual keyboard behavior on devices such as tablets, mobile phones, or other devices where a hardware keyboard may not be available, if the element's content is editable (for example, it is an {{htmlelement("input")}} or {{htmlelement("textarea")}} element, or an element with the [`contenteditable`](/en-US/docs/Web/HTML/Global_attributes/contenteditable) set).
```

> The **`virtualKeyboardPolicy`** property of the {{domxref("HTMLElement")}} interface gets and sets a string indicating the on-screen virtual keyboard behavior on devices such as tablets, mobile phones, or other devices where a hardware keyboard may not be available, if the element's content is editable (for example, it is an {{htmlelement("input")}} or {{htmlelement("textarea")}} element, or an element with the [`contenteditable`](/en-US/docs/Web/HTML/Global_attributes/contenteditable) set).

Author: ```suggestion
The **`virtualKeyboardPolicy`** property of the {{domxref("HTMLElement")}} interface gets and sets a string indicating the on-screen virtual keyboard behavior on devices such as tablets, mobile phones, or other devices where a hardware keyboard may not be available, if the element's content is editable (for example, it is an {{htmlelement("input")}} or {{htmlelement("textarea")}} element, or an element with the [`contenteditable`](/en-US/docs/Web/HTML/Global_attributes/contenteditable) attribute set).
```

> It reflects the value of the [`virtualkeyboardpolicy`](/en-US/docs/Web/HTML/Global_attributes/virtualkeyboardpolicy) HTML global attribute.

Reviewer: If I understand correctly that linked page looks wrong:

> The virtualkeyboardpolicy [global attribute](https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes) is an enumerated attribute. When specified on an element that also uses the [contenteditable](https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes#contenteditable) attribute, it controls the on-screen virtual keyboard behavior on devices such as tablets, mobile phones, or other devices where a hardware keyboard may not be available.

...implies that it is only effective if `contenteditable` is set, but is seems that it is effective for all elements whose contents are editable (i.e. including `<input>` and `<textarea>`. Is that right?

Author: yes, it is, I've also update the content of `virtualkeyboardpolicy` html global attribute

## Value

An enumerated value; possible values are:

> - `"auto"` or an empty string (`""`):
>   - The default handling of the virtualkeyboard by the browser is enabled.
> - `"manual"`:
>   - The default handling of the virtualkeyboard in a browser is prevented and the virtualkeyboard needs to be handled by the script.

Reviewer: ```suggestion
- `"auto"` or an empty string (`""`)
  - The browser automatically shows the virtual keyboard when the user taps or focuses the element.
- `"manual"`
  - The browser does not automatically show the virtual keyboard: showing/hiding the virtual keyboard is handled manually by the script.
```
